membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -49,13 +51,32 @@ object AttributesFromZuora extends LoggingWithLogstashFields {
 
     val attributesFromDynamo: Future[Option[Attributes]] = dynamoAttributeGetter(identityId)
 
-    dynamoAndZuoraAgree(attributesFromDynamo, attributes, identityId).onFailure {
+    val shouldSkipUpdate = dynamoAndZuoraAgree(attributesFromDynamo, attributesFromZuora, identityId)
+
+    shouldSkipUpdate.onFailure {
       case error => log.warn(s"Tried to compare attributes for $identityId but then ${error.getMessage}", error)
     }
 
-    attributesWithFlagFromDynamo(attributes, attributesFromDynamo)
+    val zuoraAttributesWithAdfree = attributesWithFlagFromDynamo(attributesFromZuora, attributesFromDynamo)
+
+    attributesAfterDynamoUpdate(identityId, shouldSkipUpdate, zuoraAttributesWithAdfree, dynamoAttributeUpdater)
   }
 
+  private def attributesAfterDynamoUpdate(identityId: String, skipUpdate: Future[Boolean], zuoraAttributesWithAdfree: Future[Option[Attributes]], dynamoAttributeUpdater: Attributes => Future[Either[DynamoReadError, Attributes]]) = {
+    skipUpdate flatMap { shouldSkipUpdate =>
+      if (shouldSkipUpdate)
+        zuoraAttributesWithAdfree
+      else {
+        val attributesAfterUpdate = for {
+            endAttributes <- OptionT(zuoraAttributesWithAdfree)
+          _ = dynamoAttributeUpdater(endAttributes).onFailure {
```

Comment:
this onFailure will log a failed future, but a left of error will be silently dropped

Reply:
what will happen if updates fail silently?  will we find out via sentry/cloudwatch that a high percentage of disagrees are happening?

Reply:
I'm sure there are other ways, we have quite a few places in the code where we ignore errors or don't wait for Futures!

I was just thinking whether there was a more wholistic way to realise a problem, rather than relying on spotting the failure itself.  However it's probably a lot of work and it depends what the bad stuff is if it starts failing (we would serve out of date info if zuora is down anyway?) and how likely (probably not so likely to fail silently?)  So overall not a big press to do that.